### PR TITLE
Added login and register endpoints.

### PR DESCRIPTION
**Pull request by NickZach2036**

### CODE REVIEW

File: src/endpoints.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import express, { json } from 'express';
+import { config } from 'dotenv';
+import { sign } from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+
+config();
+
+const app = express();
+const port = 3000;
```
Comment (AlexOgnyanov) [inline src/endpoints.js]: Load port from .env and assign default value if property is not present in .env (env.PORT || 3000)

File: src/endpoints.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+});
+
+const validateEmail = (email) => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+};
+
+const validatePassword = (password) => {
+  return password.length >= 6;
+};
```
Comment (AlexOgnyanov) [inline src/endpoints.js:69]: Create validateEmailOrFail and validatePasswordOrFail methods that use the validateEmail and validatePassword methods and automatically throw errors if values are not valid. Hint: res should be one of the parameters of said methods

File: src/endpoints.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  if (!user) {
+    return res.status(400).send('User not found.');
+  }
+
+  const validPassword = await bcrypt.compare(password, user.password);
+  if (!validPassword) {
+    return res.status(400).send('Incorrect password.');
+  }
+
+  const jwtSecretKey = process.env.JWT_SECRET_KEY;
```
Comment (AlexOgnyanov) [inline src/endpoints.js]: Add throw error if JWT_SECRET_KEY is not defined in .env

**Review by AlexOgnyanov — CHANGES_REQUESTED** [review]

File: src/endpoints.js (hunk 1 above)
Comment (NickZach2036) [reply src/endpoints.js]: Requested changes were acknowledged.

File: src/endpoints.js (hunk 2 above)
Comment (NickZach2036) [reply src/endpoints.js:69]: Fixed

File: src/endpoints.js (hunk 3 above)
Comment (NickZach2036) [reply src/endpoints.js]: Fixed

**Review by AlexOgnyanov — APPROVED** [review]

**Pull request merged**